Agent 3.45 (#1954)
buildkite/docs: pages/agent/v3/help/_env_set.md

Reviewer: Similar comments to the `env get` command

<!--
  _____   ____    _   _  ____ _______   ______ _____ _____ _______
 |  __ \ / __ \  | \ | |/ __ \__   __| |  ____|  __ \_   _|__   __|
 | |  | | |  | | |  \| | |  | | | |    | |__  | |  | || |    | |
 | |  | | |  | | | . ` | |  | | | |    |  __| | |  | || |    | |
 | |__| | |__| | | |\  | |__| | | |    | |____| |__| || |_   | |
 |_____/ \____/  |_| \_|\____/  |_|    |______|_____/_____|  |_|

This file is auto-generated by scripts/update-agent-help.sh, please update the
agent CLI help in https://github.com/buildkite/agent and run the generation
script.

-->

### Usage

`buildkite-agent env set [variable]`

### Description

> Sets environment variable values in the current job execution environment.  Existing variables will be overwritten.
> 
> Note that this subcommand is only available from within the job executor with the `job-api` experiment enabled.
> 
> Note that changes to the job environment variables only apply to subsequent phases of the job. To read the new values of variables from within the current phase, use `env get`.
> 
> Note that Buildkite read-only variables cannot be overwritten.

Reviewer: There's a lot of _noting_ going on here. If possible, perhaps simplify to:

```suggestion
Unsets environment variables in the current job execution environment. Changes to the job environment variables only apply to subsequent phases of the job. This command cannot unset Buildkite read-only variables.

To read the new values of variables from within the current phase, use `env get`.

Note that this subcommand is only available from within the job executor with the job-api experiment enabled.
```

> Example:

Reviewer: This should be plural like the others:

```suggestion
Examples:
```

Reviewer 2: The `Example` in `buildkite-agent env dump` is an H3, so I made these H3 too.
